src/main/java/cn/nukkit/network/protocol/AddPlayerPacket.java

```diff
@@ -32,10 +32,33 @@ public byte pid() {
     public float yaw;
     public Item item;
     public EntityMetadata metadata = new EntityMetadata();
+    
+    public int uvarint1 = 0;
+    public int uvarint2 = 0;
+    public int uvarint3 = 0; 
+    public int uvarint4 = 0;
+    
+    public int long1 = 0;
 
     @Override
     public void decode() {
-
+        this.uuid = this.getUUID();
+        this.username = this.getString();
+        this.entityUniqueId = this.getVarLong();
+        this.entityRuntimeId = this.getVarLong();
+        //x,y,z
+        //speed
```

Comment:
this.getVector3f(this.speedX, this.speedY, this.SpeedZ), need change or add one more function for that.